### DISCUSSION TITLE
Flow: Can´t delete 1:n item that went through filter hook

### DISCUSSION THREAD

**Discussion by danielband** [directus/directus#15659] [Archive]
Setup:
Postgres 13

Steps to reproduce:

1. Create a flow with an Event hook set to type filter
2. Create an item in your 1:n data model
3. Save
4. Deleting that item displays **current transaction is aborted, commands ignored until end of transaction block**

**Comment by licitdev** [comment]
What does your flow do? You should be using the `action` type event hook unless you're trying to manipulate the payload or to stop the event.

**Reply by danielband** [reply]
The flow returns the last data and as last action I had a run script that would do something like this:

```
function ({$trigger}) {
  if (SOME_CONDITION) {
    throw Error(REASON)
  }

  return $trigger.payload
}
```

Then I figured that I should cancel flows like [this](https://github.com/directus/directus/pull/15283). But that is not my issue. When the action passes the condition and returns the payload, the item gets created. Deleting that item in the child table with (the one with 1:n data model) results in  **current transaction is aborted, commands ignored until end of transaction block**.  Deleting the item in the parent table works though.

As a sidenote. I replaced the flow with a custom extension throwing InvalidPayloadException and its working just fine.

**Reply by licitdev** [reply]
Good to know the custom extension works for your use case! 👍 

If it's possible, please kindly provide clear steps for reproduction so that we can triage and resolve the issue you've faced in flows. Eg: screenshot of flow and, schema and steps undertaken. 
A screen recording would be great too. Thank you!
